client/src/frontend/containers/Login/index.js

```diff
@@ -1,12 +1,38 @@
-import React from "react";
+import { useEffect, useRef } from "react";
 import PropTypes from "prop-types";
+import { useTranslation } from "react-i18next";
 import withProjectContext from "hoc/withProjectContext";
 import SignInUp from "global/components/sign-in-up";
-import { useRouteMatch } from "react-router-dom";
+import { useRouteMatch, useLocation, useHistory } from "react-router-dom";
+import { useNotification } from "hooks";
 import * as Styled from "./styles";
 
 function LoginContainer({ projectBackLink }) {
+  const { t } = useTranslation();
   const isSignUp = useRouteMatch("/signup");
+  const location = useLocation();
+  const history = useHistory();
+
+  const notifiedRef = useRef(false);
+
+  const notifyUnauthorized = useNotification(() => ({
+    id: "authorizationError",
+    level: 2,
+    heading: "Access Denied",
+    body: t("errors.authorization"),
+    scope: "authentication"
+  }));
+
+  useEffect(() => {
+    const searchParams = new URLSearchParams(location.search);
+    const redirectUri = searchParams.get("redirect_uri");
+    const shouldNotify = !!redirectUri || !location.state?.postLoginRedirect;
+
+    if (shouldNotify && !notifiedRef.current) {
+      notifyUnauthorized();
+      notifiedRef.current = true;
+    }
```

Comment:
I want to test to see if this ref check is still necessary. I was getting maximum depth errors from notifications here at one point, but this got simplified a lot.